cine-flex/src/containers/Movies/Movies.jsx

```diff
@@ -0,0 +1,61 @@
+import Button from '../../components/Button/Button';
+import MovieCard from '../../components/MovieCard/MovieCard';
+import MovieDetail from '../../components/MovieDetail/MovieDetail';
+import styles from '../Movies/Movies.module.scss';
+import { AppConstants } from "../../constants/app-constants";
+import { useState, useEffect } from 'react';
+import { getMoviesData } from '../../services/MoviesData';
+
+/**
+ * @description - This function returns all movies container
+ * @returns 
+ */
+function Movies() {
+    const [moviesData, setMoviesData] = useState([]);
+    const [selectedMovie, setSelectedMovie] = useState(0);
+    const [noOfMovieCards, setNoOfMovieCards] = useState(6);
+    let movieCards = [];
+
+    useEffect(() => {
+        getMoviesData().then((data) => {
```

Comment:
could you let me know what happens before this promise gets resolved the component unmounted ?